spec/features/saml/loa1_sso_spec.rb

```diff
@@ -3,28 +3,35 @@
 feature 'LOA1 Single Sign On' do
   include SamlAuthHelper
 
-  context 'First time registration' do
+  context 'First time registration', email: true do
     it 'takes user to agency handoff page when sign up flow complete' do
       saml_authn_request = auth_request.create(saml_settings)
-
       visit saml_authn_request
-      sign_up_and_2fa_loa1_user
+      sp_request_id = ServiceProviderRequest.last.uuid
+      sign_up_and_2fa_loa1_user_who_came_from_sp
 
       expect(current_path).to eq sign_up_completed_path
 
       click_on t('forms.buttons.continue_to', sp: 'Your friendly Government Agency')
 
       expect(current_url).to eq saml_authn_request
+      expect(ServiceProviderRequest.from_uuid(sp_request_id)).
```

Comment:
Not that I can think of. I'm open to ideas if you have them. I want to be able to isolate the creation of the ServiceProviderRequest and then make sure it gets deleted. In a controller spec, the entire action will run in one step. AFAIK, I can't use a controller spec to verify that a particular ServiceProviderRequest was created since it will get deleted after calling the action. This controller is special in that it gets called twice, but we only want some of what it does to be called once. I think this is best tested with a feature spec.